### ISSUE TITLE
Null pointer dereference in 3.0.0-alpha1 when generating a CMAC key

### ISSUE THREAD

**Issue by sfackler** [openssl/openssl#11671]
What I think is the correct way of creating a CMAC EVP_PKEY (from the wiki: https://wiki.openssl.org/index.php/EVP_Key_and_Parameter_Generation) now hits a segfault trying to dereference the null ENGINE pointer in the EVP_PKEY_CTX:

```c
include <openssl/evp.h>

int main() {
	EVP_PKEY_CTX *kctx = EVP_PKEY_CTX_new_id(EVP_PKEY_CMAC, NULL);
	EVP_PKEY_keygen_init(kctx);
	EVP_PKEY_CTX_ctrl(kctx, -1, EVP_PKEY_OP_KEYGEN, EVP_PKEY_CTRL_CIPHER, 0, (void *)EVP_aes_256_ecb());
}
```

```
rogram terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000516c5d in ENGINE_get_id (e=0x0) at crypto/engine/eng_lib.c:251
251	    return e->id;
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.30-11.fc31.x86_64
(gdb) bt
#0  0x0000000000516c5d in ENGINE_get_id (e=0x0) at crypto/engine/eng_lib.c:251
#1  0x000000000053d34e in pkey_mac_ctrl (ctx=0x12d1080, type=12, p1=0, p2=0x70a7a0 <aesni_256_ecb>) at crypto/evp/pkey_mac.c:312
#2  0x0000000000413ce6 in EVP_PKEY_CTX_ctrl (ctx=0x12d1080, keytype=-1, optype=4, cmd=12, p1=0, p2=0x70a7a0 <aesni_256_ecb>)
    at crypto/evp/pmeth_lib.c:978
#3  0x0000000000404139 in main ()
```

**Comment by levitte** [comment]
Fix in #11674, please try it out

**Comment by sfackler** [comment]
Looks fixed to me, thanks!
